nanopub/sign_utils.py

```diff
@@ -0,0 +1,158 @@
+from base64 import decodebytes, encodebytes
+
+import requests
+from Crypto.Hash import SHA256
+from Crypto.PublicKey import RSA
+from Crypto.Signature import PKCS1_v1_5
+from rdflib import BNode, ConjunctiveGraph, Literal, Namespace, URIRef
+
+from nanopub.definitions import NANOPUB_SERVER_LIST, NP_PURL, NP_TEMP_PREFIX
+from nanopub.namespaces import NPX
+from nanopub.profile import Profile
+from nanopub.trustyuri.rdf import RdfHasher, RdfUtils
+from nanopub.trustyuri.rdf.RdfPreprocessor import transform
+from nanopub.utils import MalformedNanopubError, extract_np_metadata, log
+
+
+def add_signature(g: ConjunctiveGraph, profile: Profile, dummy_namespace: Namespace, pubinfo_uri: URIRef) -> ConjunctiveGraph:
+    """Implementation in python of the process to sign a nanopub with a RSA private key"""
+    g.add((
+        dummy_namespace["sig"],
+        NPX["hasPublicKey"],
+        Literal(profile.public_key),
+        pubinfo_uri,
+    ))
+    g.add((
+        dummy_namespace["sig"],
+        NPX["hasAlgorithm"],
+        Literal("RSA"),
+        pubinfo_uri,
+    ))
+    g.add((
+        dummy_namespace["sig"],
+        NPX["hasSignatureTarget"],
+        dummy_namespace[""],
+        pubinfo_uri,
+    ))
+    # Normalize RDF
+    quads = RdfUtils.get_quads(g)
+    normed_rdf = RdfHasher.normalize_quads(
+        quads,
+        baseuri=str(dummy_namespace),
+        hashstr=" "
+    )
+    # Note: normed_rdf needs to end with a newline
```

Comment:
Is this part of a TODO?

Reply:
Not really, just a note for developers, to not be surprised when they see an extra newline in the normed RDF